### ISSUE TITLE
Config value datatypes assumed to always be strings or functions

### ISSUE THREAD

**Issue by svan-jansson** [keathley/vapor#42]
Hi, I've noticed a strange behavior when getting config values that are not strings. Take this example:

**Here I'm setting a boolean value:**

```elixir
{:ok, true} = MyApp.Config.set("my_boolean_value",  true)
```

**I then try to access it using the `as: :bool` parameter, but this happens:**

```elixir
{:error, Vapor.ConversionError} = MyApp.Config.get("my_boolean_value", as: :bool)
```

**It will only work if i use the `as: :string` parameter, like this:**

```elixir
{:ok, true} = MyApp.Config.get("my_boolean_value", as: :string)
```
With the `as: :string` parameter I expected the returned value to always be a string, but the implementation is just a passthrough for the fetched value.

A suggestion solution is to add `get/1` and `get!/1`, without the type argument, that returns the value exactly as it is stored.

**Comment by keathley** [comment]
Thanks for the report :). I think we’re going to rethink the way we specify types. The change is described in #43

**Comment by keathley** [comment]
Will be solved in #51
